### DISCUSSION TITLE
Build fail `no matching member function for call to 'fetch_add'`

### DISCUSSION THREAD

**Discussion by frutas-fruit** [xissburg/edyn#87] [General]

**Comment by xissburg** [comment]
Perhaps it's missing an include? Sometimes things compile even though there isn't an include in the file, which it gets from another included file, but then it fails in some compilers... it's quite annoying. I see there's no `#include <atomic>` in `job_dispatcher.cpp`. You could try adding it to see what happens.

**Comment by qhdwight** [comment]
I'm getting the same issue. But from the docs, I don't see this as a function:
https://en.cppreference.com/w/cpp/atomic/atomic/fetch_add
Seems like it requires a value as first parameter?

**Reply by xissburg** [reply]
Yes, indeed. What is going on in here? 😅
I don't know why it compiles.

**Reply by qhdwight** [reply]
Hmm... well what should it be? Adding a 1? I find it funny that the code works too then, or maybe we are missing something.

**Reply by xissburg** [reply]
Yes it should have a 1 there. I will fix it in the master branch.

**Reply by xissburg** [reply]
The reason it would compile is because `std::memory_order_relaxed` is an enum value that was being treated as int and the second argument to `fetch_add` has a default value of `std::memory_order_seq_cst` so in effect zero was being passed and the variable wasn't being incremented at all. Thanks for catching this one.

Looking at it again, I am not sure that variable is still needed. The original goal was preventing the first worker from having priority when assigning the first jobs, i.e. when the worker job queue is empty, and more equally distributing jobs among workers. If you dispatch multiple jobs in sequence it might happen that the first worker gets the first job, then the second gets the second job, then when assigning the third job, it's possible the first worker is empty already so it will get the third job as well and the remaining workers won't be doing anything. Whether that's good or bad I do not know, but I just wanted to more equally distribute things. I guess I will revisit this later when I get back to work on the island worker refactor, which will allow multiple islands to exist in a single island worker.
